internal/lint/subscribe.go

```diff
@@ -22,31 +22,47 @@ type SubscriberInput struct {
 func Subscriber(input *SubscriberInput) {
 	var subject bps.Subscriber
 	var handler *mockHandler
-	var ctx = context.Background()
+	var topic string
 
-	var (
-		topic        string
-		unknownTopic string
-	)
+	var ctx context.Context
+	var cancel context.CancelFunc
 
 	ginkgo.BeforeEach(func() {
 		cycle := time.Now().UnixNano()
 		topic = fmt.Sprintf("bps-unittest-topic-%d-a", cycle)
-		unknownTopic = fmt.Sprintf("bps-unittest-topic-%d-unknown", cycle)
 
 		subject = input.Subject(topic, []bps.SubMessage{
 			bps.RawSubMessage("message-1"),
 			bps.RawSubMessage("message-2"),
 		})
 		handler = &mockHandler{}
+
+		// give consumer max 5s per test:
+		ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+	})
+
+	ginkgo.AfterEach(func() {
+		cancel()
 	})
 
 	ginkgo.It("should subscribe", func() {
-		Ω.Expect(subject.Subscribe(ctx, topic, handler)).To(Ω.Succeed())
-		Ω.Expect(extractData(handler.Received)).To(Ω.Equal([][]byte{
+		go func() {
+			defer ginkgo.GinkgoRecover()
+
+			Ω.Expect(subject.Subscribe(ctx, topic, handler)).To(Ω.Or(
+				Ω.Succeed(),
+				Ω.MatchError(context.Canceled),
+			))
+		}()
+
+		Ω.Eventually(func() interface{} {
+			return extractData(handler.Received)
```

Comment:
that's not good, because it will cause race warnings. you need to protect access to `Received`, that's why I suggested to have an additional `count uint32` that you increment atomically and read via `handler.Len()`. this is what you can use for `Eventually`. after that you can just do `Expect(handler.Received)` as we then know that it's not being written to

Reply:
keep concurrency in mind when you are using Eventually

Reply:
I will handle this in next PR - for nats.io backend (already changed this line, don't want conflicts later). Will switch to atomic counter as well.